src/dialog.jsx

```diff
@@ -262,7 +264,7 @@ const DialogInline = React.createClass({
           </TransitionItem>}
         </ReactTransitionGroup>
         <Overlay
-          show={this.props.open}
+          show={this.props.open && this.props.showOverlay}
```

Comment:
my use-case doesn't need the click away, but it does need to prevent scrolling, so I think we should just change the background-color, as @subjectix suggested